digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala

```diff
@@ -86,7 +87,14 @@ trait LinearAssetOperations {
     getByRoadLinks(typeId, roadLinks, change)
   }
 
-  private def getByRoadLinks(typeId: Int, roadLinks: Seq[RoadLink], changes: Seq[ChangeInfo]): Seq[PieceWiseLinearAsset] = {
+  private def getByRoadLinks(typeId: Int, roadLinksExist: Seq[RoadLink], changes: Seq[ChangeInfo]): Seq[PieceWiseLinearAsset] = {
+
+    var roadLinks: Seq[RoadLink] = null
```

Comment:
Maybe if we do like this is better:
    val roadLinks = typeId match {
      case LinearAssetTypes.MaintenanceRoadAssetTypeId => roadLinksExist.filter(_.functionalClass > 4)
      case _ => roadLinksExist
    }